deploy/scripts/0007-CarbonController-upgrade.ts

```diff
@@ -0,0 +1,23 @@
+import { DeployedContracts, InstanceName, setDeploymentMetadata, upgradeProxy } from '../../utils/Deploy';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+/**
+ * @dev custom trading fees upgrade
+ */
+const func: DeployFunction = async ({ getNamedAccounts }: HardhatRuntimeEnvironment) => {
+    const { deployer } = await getNamedAccounts();
+    const voucher = await DeployedContracts.Voucher.deployed();
+
+    const carbonController = await DeployedContracts.CarbonController.deployed();
+    await upgradeProxy({
+        name: InstanceName.CarbonController,
+        from: deployer,
+        args: [voucher.address, carbonController.address],
+        initImpl: true
```

Comment:
This can be a good opportunity to remove `initImpl` and initialize the implementation construct in its contstructor instead (discussed with @yudilevi a few weeks ago)

Reply:
Yup, planned as a separate PR